metrics-service/src/main/java/org/ametiste/metrics/router/MappingAggregatorsRouter.java

```diff
@@ -68,4 +75,27 @@ public MappingAggregatorsRouter(Map<String, List<MetricsAggregator>> aggregators
         return aggregatorsMap.get(DEFAULT_ROUTE_NAME);
     }
 
+    /**
+     * Matches list of aggregators either by key or by prefixed key. Example of keys:
+     * "metric.that.needs.to.be.routed" matches as to key "metric.that.needs.to.be.routed" as to
+     * "metric.that.needs*" and "*"
+     * when "metric.is.to.be.routed" matches only "*" of those route keys
+     *
+     * @param metricIdentifier - id of metric for that route is defined. Note: id is
+     *                         already resolved one, in case if {@link MetricsIdentifierResolver}
+     *                         had a match for metric identifier, identifier might be different from initial name
+     * @param function - consumer that should be applied to each of aggregators .
+     */
+    @Override
+    public void aggregate(String metricIdentifier, Consumer<MetricsAggregator> function) {
+        this.getAggregatorsForMetric(metricIdentifier).forEach(a -> {
+            try {
+                function.accept(a);
+            } catch (Exception e) {
+                logger.warn("Aggregator threw exception. Aggregator: " + a.getClass(), e);

```

Comment:
WARN level is so noisy for these purposes, DEBUG or if (isDebugEnabled()) { warn(...) } would be better.